astarte_device.c

```diff
@@ -1035,7 +1037,14 @@ static void on_incoming(
     snprintf(path, 512, "%.*s", path_len, path_begin);
 
     if (!data && data_len == 0) {
-        // TODO: handle incoming unset messages
+        if (device->unset_data_event_callback) {
+            astarte_device_unset_event_t event
+                = { .device = device, .interface_name = interface_name, .path = path };
+            device->unset_data_event_callback(&event);
+        } else {
+            ESP_LOGE(TAG,
+                "Unset data for %s received, but unset_data_event_callback is not defined", path);
```

Comment:
Let's make it shorter: `Unset for %s received, but no unset_event_callback defined.", path)`